src/main/java/duke/controller/DukeController.java

```diff
@@ -0,0 +1,167 @@
+package duke.controller;
+
+import java.io.IOException;
+
+import duke.controller.dialog.DukeDialogController;
+import duke.controller.dialog.UserDialogController;
+import duke.listener.Message;
+import duke.storage.Storage;
+import duke.task.Operation;
+import javafx.application.Platform;
+import javafx.fxml.FXML;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.ImageView;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+
+/**
+ * Controller for DukeUi. Provides the layout for the other controls.
+ */
+public class DukeController extends AnchorPane implements Message {
+    public static final String GREET = "Hello! I'm Duke.\nWhat can I do for you?";
+
+    @FXML
+    private ScrollPane chatPane;
+    @FXML
+    private VBox dialogContainer;
+    @FXML
+    private TextField tfInput;
+    @FXML
+    private ImageView ivSend;
+
+    private final Storage storage;
+
+    /**
+     * Constructs a DukeController object.
+     */
+    public DukeController() {
+        storage = new Storage(this);
+    }
+
+    /**
+     * Adds DukeDialog with content.
+     *
+     * @param content Content.
+     */
+    public void addDukeDialog(String content) {
+        if (dialogContainer == null) {
+            return;
+        }
+        try {
+            // The reason of following approach is because the size of the dukeDialogController
+            // will be returned only after it is added by the dialogContainer.
+            DukeDialogController preDukeDialogController = new DukeDialogController(content);
+            preDukeDialogController.heightProperty()
```

Comment:
Can try to break this statement up to avoid complicated statements. Also, is it optimal to have nested try-catch block? (Avoid complicated expressions under Code Quality)